UVAtlas/isochart/isochartengine.cpp

```diff
@@ -37,40 +37,50 @@ void IIsochartEngine::ReleaseIsochartEngine(
 
 CIsochartEngine::CIsochartEngine() :
     m_state(ISOCHART_ST_UNINITILAIZED),
+#ifdef WIN32
     m_hMutex(nullptr),
+#endif
     m_dwOptions(_OPTION_ISOCHART_DEFAULT)
 {
 }
 
 CIsochartEngine::~CIsochartEngine()
 {
-    // if other thread is calling public method of CIsochartEngine this time, 
+    // if other thread is calling public method of CIsochartEngine this time,
     // Free will return with "busy". So loop until free successfully.
     while (FAILED(Free()))
     {
+#ifdef WIN32
         SwitchToThread();
+#else
+        pthread_yield();
+#endif
     }
 
+#ifdef WIN32
     if (m_hMutex)
     {
         CloseHandle(m_hMutex);
     }
+#endif
 }
 
 HRESULT CIsochartEngine::CreateEngineMutex()
 {
+#ifdef WIN32
     m_hMutex = CreateMutexEx(nullptr, nullptr, CREATE_MUTEX_INITIAL_OWNER, SYNCHRONIZE);
```

Comment:
Before seeing where exactly this was used... why was this a full-on mutex instead of critical section?

Reply:
This code is quite old, so I'm not entirely sure :)